### ISSUE TITLE
labelconvert - segmentation fault with specific aparc.a2009s₊aseg file

### ISSUE THREAD

**Issue by ansk** [MRtrix3/mrtrix3#872]
Dear experts,

I encountered segmentation fault with one particular aparc.a2009s+aseg.mgz file. My command line is following:

labelconvert aparc.a2009s₊aseg.mgz $FREESURFER_HOME/FreeSurferColorLUT.txt /usr/local/mrtrix3/src/connectome/tables/fs_a2009s.txt parc_init.mif

I am attaching the problematic aparc.a2009s₊aseg file and FreeSurfer LUT file. The aparc.a2009s₊aseg file has erroneous labeling of right ventricle, therefore producing error with labelconvert is in fact good for the user, but maybe it will be better to issue more specific error than segmentation fault. Is the primary reason of segfault the fact that some tissue is non-standardly labeled as nonWM-hypointensities?

Update: I am attaching second aparc.a2009s+aseg file where labelconvert ends with segmentation fault.

Regards,

Antonin Skoch
[aparc.a2009s+aseg.mgz.tar.gz](https://github.com/MRtrix3/mrtrix3/files/680168/aparc.a2009s.aseg.mgz.tar.gz)
[aparc.a2009s+aseg.tar.gz](https://github.com/MRtrix3/mrtrix3/files/680953/aparc.a2009s.aseg.tar.gz)

[FreeSurferColorLUT.txt](https://github.com/MRtrix3/mrtrix3/files/680170/FreeSurferColorLUT.txt)

**Comment by Lestropie** [comment]
Hi @ansk,

Thanks very much for the concise report and the example data; makes our lives much easier.

The segmentation fault occurs because there's a strip of voxels in the image that contain the value -1; the possibility of which hadn't been considered when the `labelconvert` code was written. You can see the problematic region with e.g.:

```
mrcalc aparc.a2009s+aseg.mgz -1 -eq - -datatype bit | mrview aparc.a2009s+aseg.mgz -roi.load -
```

The fix is on its way in #873.

**Comment by ansk** [comment]
Dear Rob,
for your information, the negative values in aparc.a2009s+aseg were due to the bug in development version of freesurfer. This should now be fixed. Nevertheless, I think that this check you implemented in labelconvert is useful anyway.

Antonin
